pkg/appsec/appsec.go

```diff
@@ -0,0 +1,127 @@
+package appsec
+
+import (
+	"encoding/json"
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/api/apiutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NewIntakeReverseProxy returns the AppSec Intake Proxy handler according to
+// the agent configuration.
+func NewIntakeReverseProxy(transport http.RoundTripper) (http.Handler, error) {
+	disabled := func(reason string) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Add("Content-Type", "application/json")
+			w.WriteHeader(http.StatusNotImplemented)
+			if err := json.NewEncoder(w).Encode(reason); err != nil {
+				log.Error(err)
+			}
+		})
+	}
+	cfg, err := newConfig(coreconfig.Datadog)
+	if err != nil {
+		return disabled(fmt.Sprintf("appsec agent disabled due to a configuration error: %v", err)), errors.Wrap(err, "configuration: ")
+	}
+	if !cfg.Enabled {
+		log.Info("AppSec agent disabled. Set the environment variable `DD_APPSEC_ENABLED=true` or add the entry " +
```

Comment:
This sounds like we're telling the user that something is wrong and that they should take action. I think the message: "AppSec proxy disabled." should be enough information. They already know the settings since they clearly manually set it to `false` (being it enabled by default).

Reply:
Good point - fixed